test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ControllerActionExecutorTests.cs

```diff
@@ -383,24 +356,44 @@ public class ControllerActionExecutorTests
             var inputParam2 = "Second Parameter";
 
             var actionParameters = new Dictionary<string, object> { { "i", "Some Invalid Value" }, { "s", inputParam2 } };
-            var methodWithTaskOfIntReturnType = new MethodWithTaskOfIntReturnType(_controller.TaskValueTypeAction);
-            var message = TestPlatformHelper.IsMono ? "Object type {0} cannot be converted to target type: {1}" :
-                                                      "Object of type '{0}' cannot be converted to type '{1}'.";
-            var expectedException = string.Format(
-                CultureInfo.CurrentCulture,
-                message,
-                typeof(string),
-                typeof(int));
+            var methodWithTaskOfIntReturnType = new MethodWithTaskOfIntReturnType(_controller.TaskValueTypeAction);            
 
             // Act & Assert
             // If it is an unrecognized derived type we throw an InvalidOperationException.
-            var ex = await Assert.ThrowsAsync<ArgumentException>(
-                () => ControllerActionExecutor.ExecuteAsync(
-                        methodWithTaskOfIntReturnType.GetMethodInfo(),
-                        _controller,
-                        actionParameters));
+            var ex = await Assert.ThrowsAsync<InvalidCastException>(
+                () => ExecuteAction(
+                    methodWithTaskOfIntReturnType,
+                    _controller,
+                    actionParameters));
+        }
 
-            Assert.Equal(expectedException, ex.Message);
+        private async Task<object> ExecuteAction(
+            Delegate methodDelegate,
+            TestController controller,
+            IDictionary<string, object> actionParameters)
+        {
+            var executor = ObjectMethodExecutor.Create(methodDelegate.GetMethodInfo(), _controller.GetType().GetTypeInfo());
+
+            var result = await ControllerActionExecutor.ExecuteAsync(
+                                                         executor,
+                                                         controller,
+                                                         actionParameters);
+
+            return result;
+        }
+
+        private async Task<object> ExecuteAction(
+            Delegate methodDelegate,
+            TestController controller,
+            object[] actionParameters)
+        {
+            var executor = ObjectMethodExecutor.Create(methodDelegate.GetMethodInfo(), _controller.GetType().GetTypeInfo());
+
+            var result = await ControllerActionExecutor.ExecuteAsync(
+                                                         executor,
+                                                         controller,
+                                                         actionParameters);
```

Comment:
Please do a pass for style on the tests that you've added/changed. I've noticed the following things:
- Indentation (don't align things to the right, just indent one notch)
- var
- long lines
- Assert.True